Libraries/LLM/SuScaledRotaryEmbedding.swift

```diff
@@ -30,7 +30,18 @@ public class SuScaledRotaryEmbedding: Module {
         let exponent =
             MLXArray(stride(from: 0, to: dimensions, by: 2)).asType(.float32) / Float(dimensions)
         let freqs = MLX.pow(MLXArray(base), exponent)
-        self._freqs = MLXArray(longFactor).asType(.float32) * freqs
+
+        let longFactorArray = MLXArray(longFactor).asType(.float32)
+        if longFactorArray.size == 1 {
+            // If longFactor is a single value, broadcast it
+            self._freqs = MLXArray(longFactor[0]) * freqs
+        } else {
+            // If longFactor is an array, it should match the frequency dimensions
+            precondition(
+                longFactorArray.size == freqs.size,
+                "longFactor size must match frequency dimensions")
+            self._freqs = longFactorArray * freqs
+        }
```

Comment:
I don't see why this condition is necessary. Once you have the `longFactorArray` it will broadcast correctly if it has size one and if it has the wrong size it will raise an error. So you could replace it like so:

```suggestion
        self._freqs = longFactorArray * freqs
```

Reply:
In that case, it looks like no change is necessary in `SuScaledRotaryEmbedding`. I'm confused, because I think I was able to reproduce the error in https://github.com/ml-explore/mlx-swift-examples/issues/92, which now no longer seems to occur.

Reply:
Oops, I didn't notice that was how it was originally. I think you fixed #92 in https://github.com/ml-explore/mlx-swift-examples/pull/107 :) and this PR can be closed.